File: 2013-06-07/python/exercise2.py
from pyplasm import *
import random

domain = PROD([INTERVALS(20)(30),INTERVALS(21)(30)])

def height(dominio):
	x = dominio[0]
	y = dominio[1]
	z = SIN(x*y)*COS(x*y)

	if x<= 10: dz = x*y*random.random()/5000 
	else: dz = x*y*random.random()/100
	
	return [x, y, dz+z]


model = MAP(height)(domain)


lake1 = T([1,2])([2*PI-1,6*PI-1])(CIRCLE(PI/2)([12,1]))

lake2 = T([1,2,3])([2*PI+0.4,2*PI+0.8,0.25])(CIRCLE(PI/1.5)([12,1]))

lake = STRUCT([COLOR(CYAN), Q(1), lake2, lake1])

model_lake = STRUCT([model,lake])

VIEW(model_lake);

File: 2013-06-07/javascript/exercise2.js
var domain = DOMAIN([[0,20],[0,21]])([50,50])

var height  = function(dominio){
	var x = dominio[0];
	var y = dominio[1];
	var z = SIN(x*y)*COS(x*y);

	//parte pianeggiante + parte montuosa
	if(x<= 10) dz = x*y*Math.random()/500; 
	else dz = x*y*Math.random()/100;
	return [x, y, dz+z];
}

var model = COLOR([139/255,69/255,19/255])(MAP(height)(domain))

var lake1 = T([0,1])([2*PI,6*PI])(DISK(PI/2)([12,1]))

var lake2 = T([0,1])([2*PI,2*PI])(DISK(PI)([12,1]))

var lake = STRUCT([COLOR([0,180,120]), EXTRUDE([0.5]), lake2, lake1])

model_lake = STRUCT([model,lake])
DRAW(model_lake);
